pkg/antctl/handlers/agentinfo.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handlers
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"k8s.io/klog"
+
+	"github.com/vmware-tanzu/antrea/pkg/monitor"
+)
+
+var _ Factory = new(AgentInfo)
+
+// ComponentAgentInfoResponse describes the internal response struct of agent-info command.
+// The first three fields refer to connection between the agent and controller, OVSDB and
+// openflow. Subnet refers to the subnet range from the Pod CIDR allocated to the Node.
+type ComponentAgentInfoResponse struct {
```

Comment:
Is it possible to reuse the AgentInfo CRD struct?
@mengdie-song

Reply:
That covers more than what we need here and misses "subnet".

Reply:
I know it is more than what this PR supports, but does it make sense to show all in the AgentInfo CRD with this get agent-info cmd?

Reply:
I got what you mean now. Shall we just dump these important info for this milestone ? I will discuss with Mengdie about reusing CRD struct and dump all info afterwards ?

Reply:
Ok to me if we can catch the 0.3.0 release, but I do hope we revisit all CLI design items after 0.3.0.
Ideally we should keep API/CLI compatible across some number of releases, but maybe not a big deal at this stage.

Reply:
@jianjuns @lzhecheng  Sorry the late review. We can dump all the information in AntreaAgentInfo via this API. Seems this patch now focuses more on the condition information.
Maybe this CLI output can follow the struct AntreaAgentInfo, put these condition information in a subcategory like Agent Conditions?

Since CRD will only be updated every minute, the information stored in CRD is not up to date.
I would suggest to follow https://github.com/vmware-tanzu/antrea/blob/master/pkg/monitor/monitor.go#L267 to get all needed information by CLI itself.

Reply:
@jianjuns @mengdie-song Thank you. I use struct AntreaAgentInfo as response now and updated the code.